brave/src/main/java/com/linecorp/armeria/client/brave/ArmeriaHttpClientParser.java

```diff
@@ -54,32 +55,33 @@ private ArmeriaHttpClientParser() {
     @Override
     public <T> void response(HttpAdapter<?, T> rawAdapter, T res, Throwable error, SpanCustomizer customizer) {
         super.response(rawAdapter, res, error, customizer);
-        if (res instanceof RequestLog && rawAdapter instanceof ArmeriaHttpClientAdapter) {
-            final RequestLog requestLog = (RequestLog) res;
-            final ArmeriaHttpClientAdapter adapter = (ArmeriaHttpClientAdapter) rawAdapter;
-            customizer.tag(SpanTags.TAG_HTTP_HOST, adapter.authority(requestLog))
-                      .tag(SpanTags.TAG_HTTP_URL, adapter.url(requestLog))
-                      .tag(SpanTags.TAG_HTTP_PROTOCOL, adapter.protocol(requestLog));
+        if (!(res instanceof HttpClientResponse)) return;
```

Comment:
@anuraaga fyi I have an idea of how to hide this type completely, so that parsers get the "unwrapped" one. will raise a PR in a bit on brave...

Reply:
https://github.com/openzipkin/brave/pull/990